### PR TITLE
prevent twist limits being created with equal lower and upper limits

### PR DESCRIPTION
**Pull request by greerdv**
Signed-off-by: greerdv <greerdv@amazon.com>

### CODE REVIEW

File: Gems/PhysX/Code/Source/Joint/PhysXJointUtils.cpp [hunk 1]
```diff
@@ -224,8 +224,22 @@ namespace PhysX {
                 physx::PxJointLimitCone limitCone(swingLimitY, swingLimitZ);
                 joint->setSwingLimit(limitCone);
 
-                const float twistLower = AZ::DegToRad(AZStd::GetMin(configuration.m_twistLimitLower, configuration.m_twistLimitUpper));
-                const float twistUpper = AZ::DegToRad(AZStd::GetMax(configuration.m_twistLimitLower, configuration.m_twistLimitUpper));
+                float twistLower = AZ::DegToRad(AZStd::GetMin(configuration.m_twistLimitLower, configuration.m_twistLimitUpper));
+                float twistUpper = AZ::DegToRad(AZStd::GetMax(configuration.m_twistLimitLower, configuration.m_twistLimitUpper));
+                // make sure there is at least a small difference between the lower and upper limits to avoid problems in PhysX
+                const float minSwingLimitRangeRadians = AZ::DegToRad(JointConstants::MinTwistLimitRangeDegrees);
```
Comment (AMZN-alexpete) [inline Gems/PhysX/Code/Source/Joint/PhysXJointUtils.cpp:230]: would a more appropriate name here be minTwistLimitRangeRadians?

Comment (greerdv) [reply Gems/PhysX/Code/Source/Joint/PhysXJointUtils.cpp:230]: yeah, you're right, stupid typo

**Review by AMZN-alexpete — APPROVED** [review]

**Review by amzn-sean — APPROVED** [review]

File: Gems/PhysX/Code/Source/Joint/PhysXJointUtils.cpp [hunk 2]
```diff
@@ -224,8 +224,22 @@ namespace PhysX {
...
-                const float twistLower = AZ::DegToRad(AZStd::GetMin(configuration.m_twistLimitLower, configuration.m_twistLimitUpper));
-                const float twistUpper = AZ::DegToRad(AZStd::GetMax(configuration.m_twistLimitLower, configuration.m_twistLimitUpper));
+                float twistLower = AZ::DegToRad(AZStd::GetMin(configuration.m_twistLimitLower, configuration.m_twistLimitUpper));
+                float twistUpper = AZ::DegToRad(AZStd::GetMax(configuration.m_twistLimitLower, configuration.m_twistLimitUpper));
+                // make sure there is at least a small difference between the lower and upper limits to avoid problems in PhysX
+                const float minTwistLimitRangeRadians = AZ::DegToRad(JointConstants::MinTwistLimitRangeDegrees);
+                if (const float twistLimitRange = twistUpper - twistLower;
+                    twistLimitRange < minTwistLimitRangeRadians)
+                {
+                    if (twistUpper > 0.0f)
```
Comment (moraaar) [inline Gems/PhysX/Code/Source/Joint/PhysXJointUtils.cpp:234]: minor: Maybe adding a comment above `if (twistUpper > 0.0f)` would help clarifying why to choose between modifying `twistLower` or `twistUpper`.

**Review by moraaar — APPROVED** [review]

File: Gems/PhysX/Code/Source/Joint/PhysXJointUtils.cpp (hunk 2 above)
Comment (greerdv) [reply Gems/PhysX/Code/Source/Joint/PhysXJointUtils.cpp:234]: Good suggestion. The AR is nearly finished on this change, so I'll make that as a separate future change.

**Pull request merged**